pkg/distributor/distributor.go

```diff
@@ -385,10 +396,10 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				}
 
 				structuredMetadata := logproto.FromLabelAdaptersToLabels(entry.StructuredMetadata)
-				if addLogLevel && !structuredMetadata.Has(labelLevel) {
+				if addLogLevel && !hasAnyLevelLabels(structuredMetadata) {
 					logLevel := detectLogLevelFromLogEntry(entry, structuredMetadata)
 					entry.StructuredMetadata = append(entry.StructuredMetadata, logproto.LabelAdapter{
```

Comment:
should we consider not adding any data if we don't find the level ?

Reply:
> Might break queries or rules if the label is absent for some log lines at random.

I can't think of an example, this label shouldn't be in used.

Reply:
pushed the change to not set detected_level if unknown